### PR TITLE
tests/utils cleanup: remove GetRunningVirtualMachineInstanceDomainXML from vmi_cloud_init_test

### PR DESCRIPTION
**Pull request by nirdothan**
instead of obtaining the MAC and PCI addresses, that were dynamically allocated by libvirt, from the domain XML, set everything explicitly. Then we know exacly what to expect to find inside the VM. The test still asserts what it's meant to assert: that the cloudinit metadata file in the VM includes the correct tag and interface details

### What this PR does
Before this PR:
GetRunningVirtualMachineInstanceDomainXML was used in the test only in order to find out which MAC and PCI addresses were dynamically assigned, in order to validate that the cloud-init metadata file in the VM was correct
After this PR:
There no need to check the Domain XML because addresses are explicitly set  

Fixes #

### Why we need it and why it was done in this way
The following tradeoffs were made:

The following alternatives were considered:

Links to places where the discussion took place: 

### Checklist

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

### Release note

```release-note
NONE
```

### CODE REVIEW

**Comment by nirdothan** [comment]
/kind/cleanup

File: tests/vmi_cloudinit_test.go [hunk 1]
```diff
@@ -435,42 +433,33 @@ var _ = Describe("[rfe_id:151][crit:high][vendor:cnv-qe@redhat.com][level:compon
 						InterfaceBindingMethod: v1.InterfaceBindingMethod{
 							Masquerade: &v1.InterfaceMasquerade{},
 						},
+						PciAddress: "0000:01:00.0",
```
Comment (0xFelix) [inline tests/vmi_cloudinit_test.go:436]: nit: Add consts for both values

**Review by 0xFelix — COMMENTED** [review]
Great find!

/approve

File: tests/vmi_cloudinit_test.go (hunk 1 above)
Comment (nirdothan) [reply tests/vmi_cloudinit_test.go]: Absolutely. Done

**Review by 0xFelix — COMMENTED** [review]
Thanks!

/approve

**Review by RamLavi — APPROVED** [review]
nice!
/lgtm

**Pull request merged**